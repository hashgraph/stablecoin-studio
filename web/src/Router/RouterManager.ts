import type { ReverseParams } from 'named-urls';
import { reverse } from 'named-urls';
import type { History } from 'history';
import { RoutesMappingUrl } from './RoutesMappingUrl';
import { NamedRoutes } from './NamedRoutes';

export class BaseRouterManager {
	constructor(private routes: Record<NamedRoutes, string> = RoutesMappingUrl) {}

	to(
		history: History,
		namedUrl: NamedRoutes,
		params?: ReverseParams,
		state?: object,
		extra?: string,
	) {
		return history.push(reverse(`${this.routes[namedUrl]}${extra || ''}`, params), state);
	}

	getUrl(namedUrl: NamedRoutes, params?: ReverseParams, extra?: string) {
		return reverse(`${this.routes[namedUrl]}${extra || ''}`, params);
	}
}

export const RouterManager = new BaseRouterManager(RoutesMappingUrl);
