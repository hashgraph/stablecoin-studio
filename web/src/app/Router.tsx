import React from 'react';
import { Navigate, Route, Routes } from 'react-router-dom';
import Layout from '../layout/Layout';
import Dashboard from '../views/Dashboard';
import Login from '../views/Login';

interface Props {
	handleRoute: {
		component: React.FC;
		type?: string;
	};
	roleRoute: {
		component: React.FC;
		roles: string[];
	};
}

const HandleRoute = ({ component: Component, type = 'public' }: Props['handleRoute']) => {
	return <Component />;
};

const WrapperRoutesLayout = () => {
	return (
		<div>
			<Layout>
				<Routes>
					{/* Dashboard route */}
					<Route path='/dashboard' element={<Dashboard />} />

					{/* Default routes */}
					<Route path='*' element={<Navigate to='/dashboard' />} />
				</Routes>
			</Layout>
		</div>
	);
};

const Router = () => {
	return (
		<main>
			<Routes>
				{/* Public routes */}
				<Route path='/' element={<HandleRoute component={Login} />} />
				<Route path='/login' element={<HandleRoute component={Login} />} />

				{/* Private routes */}
				<Route path='/*' element={<HandleRoute component={WrapperRoutesLayout} type='private' />} />
			</Routes>
		</main>
	);
};

export default Router;
