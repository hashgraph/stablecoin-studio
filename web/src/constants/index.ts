export const CONFIG_ID_SC = '0x0000000000000000000000000000000000000000000000000000000000000002';
export const CONFIG_ID_RESERVER =
	'0x0000000000000000000000000000000000000000000000000000000000000003';
