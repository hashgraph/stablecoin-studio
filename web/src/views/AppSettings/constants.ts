export const networkOptions = [
	{ value: 'TESTNET', label: 'Tesnet' },
	{ value: 'MAINNET', label: 'Mainnet' },
];
