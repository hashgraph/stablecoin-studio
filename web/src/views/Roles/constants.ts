import { StableCoinRole } from 'hedera-stable-coin-sdk';
import type { Action } from './HandleRoles';

export const actions: Record<string, Action> = {
	edit: 'editRole' as Action,
	give: 'giveRole' as Action,
	revoke: 'revokeRole' as Action,
};

export const roleOptions = [
	{ value: StableCoinRole.CASHIN_ROLE, label: 'Cash in' },
	{ value: StableCoinRole.BURN_ROLE, label: 'Burn' },
	{ value: StableCoinRole.WIPE_ROLE, label: 'Wipe' },
	{ value: StableCoinRole.RESCUE_ROLE, label: 'Rescue' },
	{ value: StableCoinRole.PAUSER_ROLE, label: 'Pause' },
];

export const cashinLimitOptions = [
	{ value: 'INCREASE', label: 'Increase cash in limit' },
	{ value: 'DECREASE', label: 'Decrease cash in limit' },
	{ value: 'RESET', label: 'Reset cash in limit' },
	{ value: 'CHECK', label: 'Check cash in limit' },
];

export const fields = {
	amount: 'amount',
	account: 'account',
	role: 'role',
	cashinLimitOption: 'cashinLimitOption',
	name: 'name',
	supplierQuantitySwitch: 'supplierQuantitySwitch',
};
