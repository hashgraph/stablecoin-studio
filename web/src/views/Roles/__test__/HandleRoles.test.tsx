import userEvent from '@testing-library/user-event';
import HandleRoles from '../HandleRoles';
import type { Action } from '../HandleRoles';
import en from '../../../translations/en/roles.json';
import { render } from '../../../test';
import { roleOptions, fields, actions } from '../constants';
import { waitFor } from '@testing-library/react';
import type { RenderResult } from '@testing-library/react';
import { RouterManager } from '../../../Router/RouterManager';

jest.mock('../../../Router/RouterManager', () => ({
	RouterManager: {
		to: jest.fn(),
	},
}));

const translations = en.giveRole;
const validAccount = '0.0.123456';

describe(`<${HandleRoles.name} />`, () => {
	test('should render correctly on all actions', () => {
		Object.keys(actions).forEach((action) => {
			const component = render(<HandleRoles action={action as Action} />);

			expect(component.asFragment()).toMatchSnapshot(action);
		});
	});

	test('should has disabled confirm button as default', () => {
		const component = render(<HandleRoles action='giveRole' />);

		const confirmButton = component.getByTestId('confirm-btn');
		expect(confirmButton).toHaveAttribute('disabled');
	});

	test('should enable confirm button after fill form correctly', async () => {
		const component = render(<HandleRoles action='giveRole' />);

		const account = component.getByTestId(fields.account);
		userEvent.type(account, validAccount);

		const roles = component.getByTestId('select-placeholder');
		userEvent.click(roles);

		const option = component.getByText(roleOptions[0].label);
		userEvent.click(option);

		expect(roles).not.toBeInTheDocument();

		const confirmButton = component.getByTestId('confirm-btn');

		await waitFor(() => {
			expect(confirmButton).not.toHaveAttribute('disabled');
			userEvent.click(confirmButton);
		});
	});

	test('cancel button should redirect to Roles view', () => {
		const component = render(<HandleRoles action='giveRole' />);

		const anything = expect.any(Function);

		const cancelButton = component.getByTestId('cancel-btn');
		userEvent.click(cancelButton);
		expect(RouterManager.to).toHaveBeenCalledWith(anything, 'roles');
	});

	describe('Supplier role', () => {
		const fillInitialForm = (component: RenderResult) => {
			const account = component.getByTestId(fields.account);
			userEvent.type(account, validAccount);

			const roles = component.getByTestId('select-placeholder');
			userEvent.click(roles);

			const option = component.getByText('Supplier');
			userEvent.click(option);
		};

		test('should render children', async () => {
			const component = render(<HandleRoles action='giveRole' />);

			fillInitialForm(component);

			const supplerQuantity = component.getByTestId('supplier-quantity');
			expect(supplerQuantity).toBeInTheDocument();
			expect(supplerQuantity).toHaveTextContent(translations.supplierQuantityQuestion);
			expect(supplerQuantity).toHaveTextContent(translations.switchLabel);

			const switchComponent = component.getByTestId('switch');
			expect(switchComponent).toBeInTheDocument();

			const inputQuantity = component.queryByTestId('input-supplier-quantity');
			expect(inputQuantity).not.toBeInTheDocument();

			const confirmButton = component.getByTestId('confirm-btn');

			await waitFor(() => {
				expect(confirmButton).not.toHaveAttribute('disabled');
				userEvent.click(confirmButton);
			});

			expect(component.asFragment()).toMatchSnapshot('giveSupplierRole');
		});

		test('should be selected Infinity token quantity by default', () => {
			const component = render(<HandleRoles action='giveRole' />);

			fillInitialForm(component);

			const switchComponent = component.getByTestId('switch');
			expect(switchComponent).toHaveAttribute('aria-checked', 'true');
		});

		test('if set switch off then user should fill token quantity field', async () => {
			const component = render(<HandleRoles action='giveRole' />);

			fillInitialForm(component);

			const confirmButton = component.getByTestId('confirm-btn');

			await waitFor(() => {
				expect(confirmButton).not.toHaveAttribute('disabled');
			});

			const switchComponent = component.getByTestId('switch');
			userEvent.click(switchComponent);

			await waitFor(() => {
				expect(switchComponent).toHaveAttribute('aria-checked', 'false');

				expect(confirmButton).toHaveAttribute('disabled');
			});

			const inputQuantity = component.getByTestId('input-supplier-quantity');
			expect(inputQuantity).toBeInTheDocument();
			userEvent.type(inputQuantity, '12345');

			await waitFor(() => {
				expect(confirmButton).not.toHaveAttribute('disabled');
			});
		});
	});
});
