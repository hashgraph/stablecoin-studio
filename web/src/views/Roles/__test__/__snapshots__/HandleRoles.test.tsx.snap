// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HandleRoles /> Supplier role should render children: giveSupplierRole 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      aria-hidden="true"
      class="chakra-heading css-llrpa2"
      data-aria-hidden="true"
      data-testid="base-container-heading"
    >
      Role management
    </h2>
    <div
      class="css-1hkemoa"
      data-testid="base-container-children"
    >
      <div
        class="chakra-stack css-9fu93p"
      >
        <div
          class="chakra-stack css-scr4k4"
        >
          <div
            class="chakra-stack css-135a0es"
          >
            <h2
              aria-hidden="true"
              class="chakra-heading css-uig24d"
              data-aria-hidden="true"
              data-testid="title"
            >
              Give role
            </h2>
            <form
              class="chakra-stack css-b5leqj"
            >
              <div
                aria-hidden="true"
                class="chakra-stack css-1h0sumw"
                data-aria-hidden="true"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:r1h:"
                    id="field-:r1h:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        Account
                      </p>
                      <p
                        class="chakra-text css-ftb7fv"
                        data-testid="required"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-6kc61d"
                      data-testid="account"
                      id="field-:r1h:"
                      name="account"
                      placeholder="Introduce account"
                      value="0.0.123456"
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  data-testid="form_control"
                  role="group"
                >
                  <label
                    aria-hidden="true"
                    class="chakra-form__label css-1jbsrib"
                    data-aria-hidden="true"
                    for="field-:r1i:"
                    id="field-:r1i:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                        data-testid="selector-label"
                      >
                        Select role to give
                      </p>
                      <p
                        class="chakra-text css-x3tq0c"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="css-svkhat"
                  >
                    <div
                      aria-hidden="true"
                      class="css-3qrqo0"
                      data-aria-hidden="true"
                    />
                    <span
                      aria-hidden="true"
                      class="css-1f43avz-a11yText-A11yText"
                      data-aria-hidden="true"
                      id="react-select-8-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-15j38w8"
                      data-aria-hidden="true"
                    >
                      <div
                        class="css-3y2vuh"
                      >
                        <span
                          class="chakra-text css-fe5bdd"
                          data-testid="select-placeholder"
                        >
                          Select role to give
                        </span>
                        <div
                          class="css-lyj4wt"
                        >
                          Supplier
                        </div>
                        <div
                          class=" css-svnjlw"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class=" css-knqiki-input"
                            id="field-:r1i:"
                            role="combobox"
                            spellcheck="false"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class=" css-hfbj6y"
                      >
                        <div
                          class="css-1aabyhj"
                        >
                          <svg
                            class="chakra-icon css-13otjrl"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 256 256"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              fill="none"
                              height="256"
                              width="256"
                            />
                            <polyline
                              fill="none"
                              points="208 96 128 176 48 96"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="16"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      aria-hidden="true"
                      data-aria-hidden="true"
                      name="role"
                      type="hidden"
                      value="2"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="css-0"
                data-aria-hidden="true"
                data-testid="supplier-quantity"
              >
                <div
                  class="chakra-stack css-84zodg"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    How much tokens the account can mint?
                  </p>
                </div>
                <div
                  class="chakra-stack css-1wytzre"
                >
                  <p
                    class="chakra-text css-4dzt3e"
                  >
                    Infinity?
                  </p>
                  <div
                    style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 17.5px; transition: opacity 0.25s; user-select: none;"
                  >
                    <div
                      class="react-switch-bg"
                      style="height: 35px; width: 64px; margin: 0px; position: relative; background: rgb(255, 255, 255); border-radius: 17.5px; cursor: pointer; transition: background 0.25s;"
                    />
                    <div
                      class="react-switch-handle"
                      style="height: 29px; width: 29px; background: rgb(173, 244, 222); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(32px); top: 3px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
                    >
                      <div
                        style="height: 29px; width: 29px; opacity: 0; position: absolute; left: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
                      >
                        <div
                          class="chakra-stack css-191yzw1"
                        >
                          <p
                            class="chakra-text css-6ixl7s"
                            data-testid="switch-handler-no"
                          >
                            NO
                          </p>
                        </div>
                      </div>
                      <div
                        style="height: 29px; width: 29px; opacity: 1; position: absolute; left: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
                      >
                        <div
                          class="chakra-stack css-191yzw1"
                        >
                          <p
                            class="chakra-text css-6ixl7s"
                            data-testid="switch-handler-yes"
                          >
                            YES
                          </p>
                        </div>
                      </div>
                    </div>
                    <input
                      aria-checked="true"
                      checked=""
                      data-testid="switch"
                      id="switch-component"
                      role="switch"
                      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
                      type="checkbox"
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="css-ajbtx6"
          data-aria-hidden="true"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-svjswr"
          />
          <div
            class="chakra-stack css-zhuomp"
          >
            <button
              class="chakra-button css-6blmsx"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-16vubea"
              data-testid="confirm-btn"
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<HandleRoles /> should render correctly on all actions: edit 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Role management
    </h2>
    <div
      class="css-1hkemoa"
      data-testid="base-container-children"
    >
      <div
        class="chakra-stack css-9fu93p"
      >
        <div
          class="chakra-stack css-scr4k4"
        >
          <div
            class="chakra-stack css-135a0es"
          >
            <h2
              class="chakra-heading css-uig24d"
              data-testid="title"
            >
              Edit role
            </h2>
            <form
              class="chakra-stack css-b5leqj"
            >
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:r1:"
                    id="field-:r1:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        edit.accountLabel
                      </p>
                      <p
                        class="chakra-text css-ftb7fv"
                        data-testid="required"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-6kc61d"
                      data-testid="account"
                      id="field-:r1:"
                      name="account"
                      placeholder="edit.accountPlaceholder"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  data-testid="form_control"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:r2:"
                    id="field-:r2:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                        data-testid="selector-label"
                      >
                        edit.selectLabel
                      </p>
                      <p
                        class="chakra-text css-x3tq0c"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="css-svkhat"
                  >
                    <div
                      class="css-3qrqo0"
                    />
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-2-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-15j38w8"
                    >
                      <div
                        class="css-3y2vuh"
                      >
                        <span
                          class="chakra-text css-4pvr19"
                          data-testid="select-placeholder"
                        >
                          edit.selectPlaceholder
                        </span>
                        <div
                          class=" css-svnjlw"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-2-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class=" css-146k1ar-input"
                            id="field-:r2:"
                            role="combobox"
                            spellcheck="false"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class=" css-hfbj6y"
                      >
                        <div
                          class="css-1aabyhj"
                        >
                          <svg
                            class="chakra-icon css-13otjrl"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 256 256"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              fill="none"
                              height="256"
                              width="256"
                            />
                            <polyline
                              fill="none"
                              points="208 96 128 176 48 96"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="16"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="role"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="css-ajbtx6"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-svjswr"
          />
          <div
            class="chakra-stack css-zhuomp"
          >
            <button
              class="chakra-button css-6blmsx"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-16vubea"
              data-testid="confirm-btn"
              disabled=""
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<HandleRoles /> should render correctly on all actions: give 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Role management
    </h2>
    <div
      class="css-1hkemoa"
      data-testid="base-container-children"
    >
      <div
        class="chakra-stack css-9fu93p"
      >
        <div
          class="chakra-stack css-scr4k4"
        >
          <div
            class="chakra-stack css-135a0es"
          >
            <h2
              class="chakra-heading css-uig24d"
              data-testid="title"
            >
              Give role
            </h2>
            <form
              class="chakra-stack css-b5leqj"
            >
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:r9:"
                    id="field-:r9:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        give.accountLabel
                      </p>
                      <p
                        class="chakra-text css-ftb7fv"
                        data-testid="required"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-6kc61d"
                      data-testid="account"
                      id="field-:r9:"
                      name="account"
                      placeholder="give.accountPlaceholder"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  data-testid="form_control"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:ra:"
                    id="field-:ra:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                        data-testid="selector-label"
                      >
                        give.selectLabel
                      </p>
                      <p
                        class="chakra-text css-x3tq0c"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="css-svkhat"
                  >
                    <div
                      class="css-3qrqo0"
                    />
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-3-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-15j38w8"
                    >
                      <div
                        class="css-3y2vuh"
                      >
                        <span
                          class="chakra-text css-4pvr19"
                          data-testid="select-placeholder"
                        >
                          give.selectPlaceholder
                        </span>
                        <div
                          class=" css-svnjlw"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-3-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class=" css-146k1ar-input"
                            id="field-:ra:"
                            role="combobox"
                            spellcheck="false"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class=" css-hfbj6y"
                      >
                        <div
                          class="css-1aabyhj"
                        >
                          <svg
                            class="chakra-icon css-13otjrl"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 256 256"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              fill="none"
                              height="256"
                              width="256"
                            />
                            <polyline
                              fill="none"
                              points="208 96 128 176 48 96"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="16"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="role"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="css-ajbtx6"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-svjswr"
          />
          <div
            class="chakra-stack css-zhuomp"
          >
            <button
              class="chakra-button css-6blmsx"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-16vubea"
              data-testid="confirm-btn"
              disabled=""
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<HandleRoles /> should render correctly on all actions: revoke 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Role management
    </h2>
    <div
      class="css-1hkemoa"
      data-testid="base-container-children"
    >
      <div
        class="chakra-stack css-9fu93p"
      >
        <div
          class="chakra-stack css-scr4k4"
        >
          <div
            class="chakra-stack css-135a0es"
          >
            <h2
              class="chakra-heading css-uig24d"
              data-testid="title"
            >
              Revoke role
            </h2>
            <form
              class="chakra-stack css-b5leqj"
            >
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:rh:"
                    id="field-:rh:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        revoke.accountLabel
                      </p>
                      <p
                        class="chakra-text css-ftb7fv"
                        data-testid="required"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-6kc61d"
                      data-testid="account"
                      id="field-:rh:"
                      name="account"
                      placeholder="revoke.accountPlaceholder"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  data-testid="form_control"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1jbsrib"
                    for="field-:ri:"
                    id="field-:ri:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                        data-testid="selector-label"
                      >
                        revoke.selectLabel
                      </p>
                      <p
                        class="chakra-text css-x3tq0c"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="css-svkhat"
                  >
                    <div
                      class="css-3qrqo0"
                    />
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-4-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-15j38w8"
                    >
                      <div
                        class="css-3y2vuh"
                      >
                        <span
                          class="chakra-text css-4pvr19"
                          data-testid="select-placeholder"
                        >
                          revoke.selectPlaceholder
                        </span>
                        <div
                          class=" css-svnjlw"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-4-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class=" css-146k1ar-input"
                            id="field-:ri:"
                            role="combobox"
                            spellcheck="false"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class=" css-hfbj6y"
                      >
                        <div
                          class="css-1aabyhj"
                        >
                          <svg
                            class="chakra-icon css-13otjrl"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 256 256"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              fill="none"
                              height="256"
                              width="256"
                            />
                            <polyline
                              fill="none"
                              points="208 96 128 176 48 96"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="16"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="role"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="css-ajbtx6"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-svjswr"
          />
          <div
            class="chakra-stack css-zhuomp"
          >
            <button
              class="chakra-button css-6blmsx"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-16vubea"
              data-testid="confirm-btn"
              disabled=""
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
